components/net/http_loader.rs

```diff
@@ -674,8 +674,10 @@ pub fn http_redirect_fetch(request: &mut Request,
     request.redirect_count += 1;
 
     // Step 7
-    // FIXME: Correctly use request's origin
-    let same_origin = location_url.origin() == request.current_url().origin();
+    let same_origin = match request.origin {
+        Origin::Origin(ref origin) => *origin == location_url.origin(),
+        Origin::Client => false,
```

Comment:
Why is this correct?

https://fetch.spec.whatwg.org/#concept-request-origin says you are not supposed to have a "client" value at this point in time, so at the very least that FIXME needs to stay.

Reply:
Because every time we create a request, we make sure to initialize it with the global's origin, so that the origin can never be client.

Reply:
Make it panic, then.